app/components/RegionDropdown/RegionDropdown.tsx

```diff
@@ -16,7 +17,7 @@ const options: Option[] = [
 ];
 
 type RegionDropdownProps = {
-  selectedRegion: RegionOptionId;
+  selectedRegion: RegionOptionId | string | undefined;
```

Comment:
Since Region option id is coming from query params it can be any given string (or undefined)